lib/model.js

```diff
@@ -4675,12 +4678,13 @@ function populate(model, docs, options, callback) {
   }
 
   const modelsMap = getModelsMapForPopulate(model, docs, populateOptions);
+  
   if (modelsMap instanceof MongooseError) {
     return immediate(function() {
       callback(modelsMap);
     });
   }
-
+  model.schema.virtuals[populateOptions.path].options.localField = originalLocalField;
```

Comment:
i think the same would be true here: add a check against "options" being undefined